distributed/_concurrent_futures_thread.py

```diff
@@ -50,6 +50,28 @@ def _python_exit():
 atexit.register(_python_exit)
 
 
+class WorkerThreadInterrupt(threading.ThreadError):
+    pass
+
+
+def _async_raise(tid: int, exctype: type = WorkerThreadInterrupt):
```

Comment:
Maybe `_set_thread_exception`? The async prefix is a bit misleading (I know the proper ABI uses it) since it is mixing with asyncio async/coroutines.